### PR TITLE
refactor RBACs to use different sa per component

### PR DESCRIPTION
**Pull request by matthyx**

### CODE REVIEW

**Comment by matthyx** [comment]
@Daniel-GrunbergerCA let me resolve the merge conflict and then you can try with the system test

**Comment by Daniel-GrunbergerCA** [comment]
Ok
@matthyx pls lmk when I  can test

File: charts/kubescape-cloud-operator/templates/kollector/clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+kind: ClusterRole
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/kollector/clusterrole.yaml:1]: I think the kollector listens only to the Kubescape namespace objects, so we should consider using `Role` instead

File: charts/kubescape-cloud-operator/templates/kubevuln/serviceaccount.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.createKubescapeServiceAccount -}}
+{{- if and .Values.kubevuln.enabled }}
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/kubevuln/serviceaccount.yaml:1]:
```suggestion
{{- if .Values.kubevuln.enabled }}
```

File: charts/kubescape-cloud-operator/templates/operator/role.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+{{- if and .Values.operator.enabled }}
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ .Values.operator.name }}
+  namespace: {{ .Values.ksNamespace }}
+rules:
+  - apiGroups: ["spdx.softwarecomposition.kubescape.io"]
+    resources: ["sbomspdxv2p3s"]
+    verbs: ["get", "watch", "list"]
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/operator/role.yaml:10]:
```suggestion
    verbs: ["get", "watch", "list", "delete"]
```

File: charts/kubescape-cloud-operator/templates/operator/role.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  - apiGroups: ["spdx.softwarecomposition.kubescape.io"]
+    resources: ["sbomspdxv2p3filtereds"]
+    verbs: ["get", "watch", "list"]
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/operator/role.yaml:13]:
```suggestion
    verbs: ["get", "watch", "list", "delete"]
```

File: charts/kubescape-cloud-operator/templates/operator/role.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{{- if and .Values.operator.enabled }}
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ .Values.operator.name }}
+  namespace: {{ .Values.ksNamespace }}
+rules:
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/operator/role.yaml:7]: We should add the vulnerabilities CRDs as well

File: charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml [hunk 6]
```diff
@@ -3,28 +3,28 @@
 apiVersion: apps/v1
 kind: DaemonSet
 metadata:
-  name: {{ .Values.nodeAgent.daemonset.name }}
+  name: {{ .Values.nodeAgent.name }}
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml:6]: I did not see the related change in the values file

**Review by dwertent — CHANGES_REQUESTED** [review]

File: charts/kubescape-cloud-operator/templates/kollector/clusterrole.yaml (hunk 1 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/kollector/clusterrole.yaml:1]: but didn't we want it to send us the size of the cluster, etc... ?

Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/kollector/clusterrole.yaml:1]: I think we need clusterrole for all these watches:
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/namespacewatch.go#L30
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/secretwatch.go#L35
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/cronjobwatch.go#L29
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/servicewatch.go#L33
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/nodewatch.go#L93
https://github.com/kubescape/kollector/blob/25cee9cf34cdd6e7af1cd0d11dc19fc329062359/watch/podwatch.go#L84

File: charts/kubescape-cloud-operator/templates/kubescape/clusterrole.yaml [hunk 7]
```diff
@@ -1,15 +1,14 @@
 kind: ClusterRole
```
[the remark this answers is not in the document]
Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/kubescape/clusterrole.yaml:1]: I have added a bunch, until it stops crying

File: charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml (hunk 6 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/node-agent/daemonset.yaml:6]: https://github.com/kubescape/helm-charts/pull/196/files#diff-233118cd3763c2ae09c5170bc1382aff6de42604596e47881db3b11672f21a31R565

File: charts/kubescape-cloud-operator/templates/operator/role.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+{{- if .Values.operator.enabled }}
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ .Values.operator.name }}
+  namespace: {{ .Values.ksNamespace }}
+rules:
```
Comment (dwertent) [inline charts/kubescape-cloud-operator/templates/operator/role.yaml:7]: The Operator needs the ability to create and delete cronjobs, configmaps, and secrets in the Kubescape namespace

**Review by dwertent — CHANGES_REQUESTED** [review]

File: charts/kubescape-cloud-operator/templates/operator/role.yaml (hunk 8 above)
Comment (matthyx) [reply charts/kubescape-cloud-operator/templates/operator/role.yaml:7]: added

**Comment by dwertent** [comment]
@matthyx Here is what I see in the Kubescape logs when triggering the host scanner:
```
{"level":"error","ts":"2023-06-15T06:05:42Z","msg":"failed to init host scanner","error":"failed to apply host scanner YAML, reason: failed to create/update YAML, reason: failed to CREATE resource, workload: '{\"apiVersion\":\"apps/v1\",\"kind\":\"DaemonSet\",\"metadata\":{\"labels\":{\"app\":\"host-scanner\",\"k8s-app\":\"kubescape-host-scanner\"},\"name\":\"host-scanner\",\"namespace\":\"kubescape\"},\"spec\":{\"selector\":{\"matchLabels\":{\"name\":\"host-scanner\"}},\"template\":{\"metadata\":{\"labels\":{\"name\":\"host-scanner\",\"otel\":\"enabled\"}},\"spec\":{\"automountServiceAccountToken\":false,\"containers\":[{\"env\":[{\"name\":\"KS_LOGGER_LEVEL\",\"value\":\"info\"},{\"name\":\"KS_LOGGER_NAME\",\"value\":\"zap\"},{\"name\":\"ACCOUNT_ID\",\"value\":\"55014013-feac-4012-a8eb-e81461ba0961\"},{\"name\":\"CLUSTER_NAME\",\"value\":\"gke_armo-test-clusters_us-central1-c_dwertent-playground\"},{\"name\":\"OTEL_COLLECTOR_SVC\",\"value\":\"otel-collector.kubescape.svc:4317\"}],\"image\":\"quay.io/kubescape/host-scanner:v1.0.61\",\"livenessProbe\":{\"httpGet\":{\"path\":\"/healthz\",\"port\":7888},\"periodSeconds\":10},\"name\":\"host-sensor\",\"ports\":[{\"containerPort\":7888,\"name\":\"scanner\",\"protocol\":\"TCP\"}],\"resources\":{\"limits\":{\"cpu\":\"0.4m\",\"memory\":\"400Mi\"},\"requests\":{\"cpu\":\"0.1m\",\"memory\":\"200Mi\"}},\"securityContext\":{\"allowPrivilegeEscalation\":true,\"privileged\":true,\"procMount\":\"Unmasked\",\"readOnlyRootFilesystem\":true},\"startupProbe\":{\"failureThreshold\":30,\"httpGet\":{\"path\":\"/readyz\",\"port\":7888},\"periodSeconds\":1},\"volumeMounts\":[{\"mountPath\":\"/host_fs\",\"name\":\"host-filesystem\"}]}],\"dnsPolicy\":\"ClusterFirstWithHostNet\",\"hostIPC\":true,\"hostPID\":true,\"terminationGracePeriodSeconds\":120,\"tolerations\":[{\"effect\":\"NoSchedule\",\"key\":\"node-role.kubernetes.io/control-plane\",\"operator\":\"Exists\"},{\"effect\":\"NoSchedule\",\"key\":\"node-role.kubernetes.io/master\",\"operator\":\"Exists\"}],\"volumes\":[{\"hostPath\":{\"path\":\"/\",\"type\":\"Directory\"},\"name\":\"host-filesystem\"}]}}}}', reason: daemonsets.apps is forbidden: User \"system:serviceaccount:kubescape:kubescape\" cannot create resource \"daemonsets\" in API group \"apps\" in the namespace \"kubescape\""}
```

**Comment by matthyx** [comment]
> @matthyx Here is what I see in the Kubescape logs when triggering the host scanner:
> 
> ```
> {"level":"error","ts":"2023-06-15T06:05:42Z","msg":"failed to init host scanner","error":"failed to apply host scanner YAML, reason: failed to create/update YAML, reason: failed to CREATE resource, workload: '{\"apiVersion\":\"apps/v1\",\"kind\":\"DaemonSet\",\"metadata\":{\"labels\":{\"app\":\"host-scanner\",\"k8s-app\":\"kubescape-host-scanner\"},\"name\":\"host-scanner\",\"namespace\":\"kubescape\"},\"spec\":{\"selector\":{\"matchLabels\":{\"name\":\"host-scanner\"}},\"template\":{\"metadata\":{\"labels\":{\"name\":\"host-scanner\",\"otel\":\"enabled\"}},\"spec\":{\"automountServiceAccountToken\":false,\"containers\":[{\"env\":[{\"name\":\"KS_LOGGER_LEVEL\",\"value\":\"info\"},{\"name\":\"KS_LOGGER_NAME\",\"value\":\"zap\"},{\"name\":\"ACCOUNT_ID\",\"value\":\"55014013-feac-4012-a8eb-e81461ba0961\"},{\"name\":\"CLUSTER_NAME\",\"value\":\"gke_armo-test-clusters_us-central1-c_dwertent-playground\"},{\"name\":\"OTEL_COLLECTOR_SVC\",\"value\":\"otel-collector.kubescape.svc:4317\"}],\"image\":\"quay.io/kubescape/host-scanner:v1.0.61\",\"livenessProbe\":{\"httpGet\":{\"path\":\"/healthz\",\"port\":7888},\"periodSeconds\":10},\"name\":\"host-sensor\",\"ports\":[{\"containerPort\":7888,\"name\":\"scanner\",\"protocol\":\"TCP\"}],\"resources\":{\"limits\":{\"cpu\":\"0.4m\",\"memory\":\"400Mi\"},\"requests\":{\"cpu\":\"0.1m\",\"memory\":\"200Mi\"}},\"securityContext\":{\"allowPrivilegeEscalation\":true,\"privileged\":true,\"procMount\":\"Unmasked\",\"readOnlyRootFilesystem\":true},\"startupProbe\":{\"failureThreshold\":30,\"httpGet\":{\"path\":\"/readyz\",\"port\":7888},\"periodSeconds\":1},\"volumeMounts\":[{\"mountPath\":\"/host_fs\",\"name\":\"host-filesystem\"}]}],\"dnsPolicy\":\"ClusterFirstWithHostNet\",\"hostIPC\":true,\"hostPID\":true,\"terminationGracePeriodSeconds\":120,\"tolerations\":[{\"effect\":\"NoSchedule\",\"key\":\"node-role.kubernetes.io/control-plane\",\"operator\":\"Exists\"},{\"effect\":\"NoSchedule\",\"key\":\"node-role.kubernetes.io/master\",\"operator\":\"Exists\"}],\"volumes\":[{\"hostPath\":{\"path\":\"/\",\"type\":\"Directory\"},\"name\":\"host-filesystem\"}]}}}}', reason: daemonsets.apps is forbidden: User \"system:serviceaccount:kubescape:kubescape\" cannot create resource \"daemonsets\" in API group \"apps\" in the namespace \"kubescape\""}
> ```
> 
> In other words, Kubescape will need permission to create and delete deamonset in the kubescape namespace

I am also dropping the role to create a namespace, because it's not kubescape's responsibility to do it.

**Comment by dwertent** [comment]
@matthyx Now there are conflicts :(

**Comment by matthyx** [comment]
I'll fix them later

**Review by dwertent — APPROVED** [review]

**Pull request merged**